extensions/db/src/main/java/org/mule/extension/db/internal/domain/executor/StoredProcedureExecutor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.db.internal.domain.executor;
+
+import org.mule.extension.db.internal.domain.autogeneratedkey.AutoGeneratedKeyStrategy;
+import org.mule.extension.db.internal.domain.autogeneratedkey.NoAutoGeneratedKeyStrategy;
+import org.mule.extension.db.internal.domain.connection.DbConnection;
+import org.mule.extension.db.internal.domain.query.Query;
+import org.mule.extension.db.internal.domain.statement.StatementFactory;
+import org.mule.extension.db.internal.result.statement.StatementResultHandler;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Executes stored procedure queries
+ */
+public class StoredProcedureExecutor extends AbstractSingleQueryExecutor {
+
+  private final StatementResultHandler statementResultHandler;
+
+  public StoredProcedureExecutor(StatementFactory statementFactory, StatementResultHandler statementResultHandler) {
+    super(statementFactory);
+    this.statementResultHandler = statementResultHandler;
+  }
+
+  @Override
+  protected Object doExecuteQuery(DbConnection connection, Statement statement, Query query) throws SQLException {
+    return doExecuteQuery(connection, statement, query, new NoAutoGeneratedKeyStrategy());
+  }
+
+  @Override
+  protected Object doExecuteQuery(DbConnection connection, Statement statement, Query query,
+                                  AutoGeneratedKeyStrategy autoGeneratedKeyStrategy)
+      throws SQLException {
+    autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
```

Comment:
Browsing through the code I see closing of the statements in different places in the code. I understand that when you want to stream results it isn't easy to centralize this. Still I think some effort should be invested in centralizing the closing and releasing of resources.
For example a background thread could be started which awaits for a CountDownLatch to reach zero. When it reaches zero it could close and release resources.
When you start this thread in execute in AbstractSingleQueryExecutor you have a way of centralizing this. Maybe I expect a interface containing methods like "awaitCompletion", "completed",...